thingsboard_gateway/extensions/tcpserver/tcpserver_connector.py

```diff
@@ -0,0 +1,243 @@
+                for attribute_updated in content["data"]:
+                    if attribute_updates_command_config["tag"] == attribute_updated:
+                        to_process = {
+                            "device": content["device"],
+                            "data": {
+                                "method": attribute_updated,
+                                "params": content["data"][attribute_updated]
+                            }
+                        }
+                        self.__process_rpc_request(to_process, attribute_updates_command_config)
+        except Exception as e:
+            log.exception(e)
+
+    def server_side_rpc_handler(self, content):
+        try:
+            if content.get("device") is not None:
+
+                log.debug("TCPServer connector received rpc request for %s with content: %s", content["device"], content)
+                if isinstance(self.__devices[content["device"]]["config"]["rpc"], dict):
+                    rpc_command_config = self.__devices[content["device"]]["config"]["rpc"].get(content["data"]["method"])
+                    if rpc_command_config is not None:
+                        self.__process_rpc_request(content, rpc_command_config)
+                elif isinstance(self.__devices[content["device"]]["config"]["rpc"], list):
+                    for rpc_command_config in self.__devices[content["device"]]["config"]["rpc"]:
+                        if rpc_command_config["tag"] == content["data"]["method"]:
+                            self.__process_rpc_request(content, rpc_command_config)
+                            break
+                else:
+                    log.error("Received rpc request, but method %s not found in config for %s.",
+                              content["data"].get("method"),
+                              self.get_name())
+                    self.__gateway.send_rpc_reply(content["device"],
+                                                  content["data"]["id"],
+                                                  {content["data"]["method"]: "METHOD NOT FOUND!"})
+            else:
+                log.debug("Received RPC to connector: %r", content)
+        except Exception as e:
+            log.exception(e)
+
+    def __process_rpc_request(self, content, rpc_command_config):
+        if rpc_command_config is not None:
+            print("content:", content)
+            send_data = ""
+            if content['data']['method'] == 'setValue':
+                if content["data"]["params"]:
+                    send_data = rpc_command_config['onCommand']
+                else:
+                    send_data = rpc_command_config['offCommand']
+            if content['data']['method'] == 'openAlarm':
+                print(content['data']['params'])
+                vol = content['data']['params']['vol']
+                song = content['data']['params']['song']
+                send_data = "7EFF06080000" + song + "EF"
```

Comment:
Please move command template (parameters and base command) into the configuration file.

Reply:
ok